test/RevenuePool-test.js

```diff
@@ -797,6 +797,109 @@ contract("RevenuePool", (accounts) => {
+        { from: relayer }
+      );
+      let merchantCreation = await getParamsFromEvent(
+        merchantTx,
+        eventABIs.MERCHANT_CREATION,
+        merchantManager.address
+      );
+      merchantSafe = merchantCreation[0]["merchantSafe"];
+    });
+
+    afterEach(async () => {
+      await merchantManager.enableSafe(merchantSafe);
+      expect(await merchantManager.getDisabledMerchantSafes()).to.have.empty;
+    });
+
+    it("allows owner to disable merchant safe", async () => {
+      await merchantManager.disableSafe(merchantSafe, { from: owner });
+      expect(await merchantManager.getDisabledMerchantSafes()).to.have.members([
+        merchantSafe,
+      ]);
+    });
+
+    it("allows registrar to disable merchant safe", async () => {
+      await merchantManager.disableSafe(merchantSafe, { from: relayer });
+      expect(await merchantManager.getDisabledMerchantSafes()).to.have.members([
+        merchantSafe,
+      ]);
+    });
+
+    it("does not allow non-owner nor non-registrar to disable merchant safe", async () => {
+      await merchantManager
+        .disableSafe(merchantSafe, { from: accounts[11] })
+        .should.be.rejectedWith(
+          Error,
+          // the real revert reason is behind the gnosis safe execTransaction
+          // boundary, so we just get this generic error
+          "caller is not an owner nor a registrar"
+        );
+    });
+  });
+
+  describe("enable merchant safe", () => {
+    let merchantSafe;
+    before(async () => {
+      await merchantManager.setup(
+        actionDispatcher.address,
+        gnosisSafeMasterCopy.address,
+        proxyFactory.address,
+        [relayer],
+        versionManager.address
+      );
+
+      let merchant = accounts[13];
+      let merchantTx = await merchantManager.registerMerchant(
+        merchant,
+        "did:cardstack:56d6fc54-d399-443b-8778-d7e4512d3a49-xz",
+        { from: relayer }
+      );
+      let merchantCreation = await getParamsFromEvent(
+        merchantTx,
+        eventABIs.MERCHANT_CREATION,
+        merchantManager.address
+      );
+      merchantSafe = merchantCreation[0]["merchantSafe"];
+    });
+
+    it("allows owner to enable merchant safe", async () => {
+      await merchantManager.enableSafe(merchantSafe, { from: owner });
+      expect(await merchantManager.getDisabledMerchantSafes()).to.have.empty;
+    });
+
+    it("allows registrar to enable merchant safe", async () => {
+      await merchantManager.enableSafe(merchantSafe, { from: relayer });
+      expect(await merchantManager.getDisabledMerchantSafes()).to.have.empty;
+    });
+
+    it("does not allow non-owner nor non-registrar to enable merchant safe", async () => {
+      await merchantManager
+        .enableSafe(merchantSafe, { from: accounts[11] })
+        .should.be.rejectedWith(
+          Error,
+          // the real revert reason is behind the gnosis safe execTransaction
+          // boundary, so we just get this generic error
```

Comment:
This comment is actually incorrect here, the real revert reason is present I think

Reply:
Yeah the comment is wrong, we don't use a gnosis safe transaction for this function.

Reply:
removed the comment